DESIGN.md

```diff
@@ -0,0 +1,156 @@
+# Design of nupm :warning: Work In Progress :warning: 
+
+This file collects design ideas and directions. The intention is iterate on this document by PRs with discussion.
+
+> **Note**  
+> in the following, until we settle down on precise names, we use the following placeholders:
+> - `METADATA_FILE`: the file containing the metadata of a package,
```

Comment:
Are we going to prepare some `METADATA_REFERENCE.md` in this PR, too? 

I know, emergent design is cool, but we already know a bit about what can be useful in a package manifest/metadata. I would advise against postponing the metadata design.

Reply:
Yes, we can sketch it out, then implement piece by piece as we need the additional functionality.

Reply:
added in `docs/design/references/METADATA.md` in 8fa88e65763350fb622e91db59ff6284b5cdf7dc

Reply:
I think that I agree with @kubouch and @amtoine (Discord) to postpone (or discard completely) the pre-/post- hooks functionality. I'm totally fine with removing it from this file. Besides — 👍🏻